content/en/blog/_posts/2024-08-14-Kubernetes-v1-31-Release.md

```diff
@@ -0,0 +1,289 @@
+    
+
+### Graduations to Stable
+
+This lists all the features that graduated to stable (also known as _general availability_). For a full list of updates including new features and graduations from alpha to beta, see the release notes.
+
+This release includes a total of 11 enhancements promoted to Stable:
+
+* [PersistentVolume last phase transition time](	https://github.com/kubernetes/enhancements/issues/3762)
+* [Metric cardinality enforcement](	https://github.com/kubernetes/enhancements/issues/2305)
+* [Kube-proxy improved ingress connectivity reliability](	https://github.com/kubernetes/enhancements/issues/3836)
+* [Add CDI devices to device plugin API](	https://github.com/kubernetes/enhancements/issues/4009)
+* [Move cgroup v1 support into maintenance mode](	https://github.com/kubernetes/enhancements/issues/4569)
+* [AppArmor support](	https://github.com/kubernetes/enhancements/issues/24)
+* [PodHealthyPolicy for PodDisruptionBudget](	https://github.com/kubernetes/enhancements/issues/3017)
+* [Retriable and non-retriable Pod failures for Jobs](	https://github.com/kubernetes/enhancements/issues/3329)
+* [Elastic Indexed Jobs](	https://github.com/kubernetes/enhancements/issues/3715)
+* [Allow StatefulSet to control start replica ordinal numbering](	https://github.com/kubernetes/enhancements/issues/3335)
+* [Random Pod selection on ReplicaSet downscaling](	https://github.com/kubernetes/enhancements/issues/2185)
+    
+    
+### Deprecations and Removals 
+
+Also see our [Kubernetes Removals and Major Changes In v1.31](https://kubernetes.io/blog/2024/07/19/kubernetes-1-31-upcoming-changes/) blog.
+    
+#### Cgroup v1 enters the maintenance mode
+
+As Kubernetes continues to evolve and adapt to the changing landscape of container orchestration, the community has decided to move cgroup v1 support into maintenance mode in v1.31.
+This shift aligns with the broader industry's move towards [cgroup v2](https://kubernetes.io/docs/concepts/architecture/cgroups/), offering improved functionality, scalability, and a more consistent interface. 
+Maintance mode means that no new features will be added to cgroup v1 support. 
+Critical security fixes will still be provided, however, bug-fixing is now best-effort, meaning major bugs may be fixed if feasible, but some issues might remain unresolved.
+
+Its recommended that you start switching to use cgroup v2 as soon as possible. For more details on the reasoning behind placing cgroup v1 in maintaince mode and starting the migration process see the [Moving cgroup v1 Support into Maintenance Mode in Kubernetes 1.31](/blog/kubernetes-1-31-moving-cgroup-v1-support-maintenance-mode) article. 
+
+Please report any problems encounter by filing an [issue](https://github.com/kubernetes/kubernetes).
+
+This work was done as part of [KEP #4569](https://github.com/kubernetes/enhancements/issues/4569) by [SIG Node](https://github.com/kubernetes/community/tree/master/sig-node).
+
```

Comment:
Please add a new removal chapter with something like this (feel free to rephrase):

```suggestion

####  InTreePluginXXXUnregister feature gates removed.

Alpha feature gates `InTreePluginAWSUnregister`, `InTreePluginAzureDiskUnregister`, `InTreePluginAzureFileUnregister`, `InTreePluginGCEUnregister`, `InTreePluginOpenStackUnregister` and `InTreePluginvSphereUnregister` were removed in Kubernetes 1.31.

These feature gates allowed us to test situation when the in-tree volume plugins were removed from code without actually removing them. The mentioned volume plugins were removed already in Kubernetes 1.30 and these feature gates did nothing useful.

The only remaining CSI migration gate is `InTreePluginPortworxUnregister` and it remains alpha until the CSI migration for Portworx is complete and the in-tree volume plugin is ready for removal.

```

* PR that removed the feature gates: https://github.com/kubernetes/kubernetes/pull/124815 
* Issue about missing documentation: https://github.com/kubernetes/kubernetes/issues/126406

Reply:
This is a possible version

```
In Kubernetes 1.31, the following alpha feature gates `InTreePluginAWSUnregister`, `InTreePluginAzureDiskUnregister`, `InTreePluginAzureFileUnregister`, `InTreePluginGCEUnregister`, `InTreePluginOpenStackUnregister`, and `InTreePluginvSphereUnregister` have been removed.

These feature gates were introduced to facilitate the testing of scenarios where in-tree volume plugins were removed from the codebase, without actually removing them. Since Kubernetes 1.30 had deprecated these in-tree volume plugins, these feature gates were redundant and no longer served a purpose.

The only CSI migration gate still standing is `InTreePluginPortworxUnregister`, which will remain in alpha until the CSI migration for Portworx is completed and its in-tree volume plugin will be ready for removal.
```

Reply:
@jsafrane I've added in this information, can you review

Reply:
Looks good to me, thanks!

Reply:
thanks @jsafrane and @mbianchidev resolving this comment :)